core/src/main/java/com/linecorp/armeria/common/util/TimeoutController.java

```diff
@@ -45,17 +49,24 @@
      *
      * @param newTimeoutMillis the new timeout value in milliseconds. {@code 0} if disabled.
      * @return {@code true} if the current timeout is reset by the specified {@code newTimeoutMillis}.
+     *         {@code false} if the current timeout could not be reset
+     *         because of some constraints in the class that implements this interface.
      */
     boolean resetTimeout(long newTimeoutMillis);
 
     /**
      * Trigger the current timeout immediately.
+     * @return {@code true} if the current timeout is triggered successfully.
+     *         {@code false} if the current timeout could not be triggered now because
+     *         the timeout schedule has already completed normally.
      */
-    void timeoutNow();
+    boolean timeoutNow();
 
     /**
      * Cancels the current timeout scheduled.
-     * @return {@code true} if the current timeout is canceled.
+     * @return {@code true} if the current timeout is canceled or no timeout is being scheduled.
+     *         {@code false} if the current timeout could not be cancelled typically because
+     *         the timeout schedule has already completed normally.
```

Comment:
.. if the timeout has been triggered already?

What happens if no timeout was scheduled previously? What happens if a user calls this method and then scheduled a new timeout?

Reply:
> What happens if no timeout was scheduled previously?

It should return `true`. Let me rewrite the `no timeout is being scheduled` to `no timeout was scheduled previously`.

> What happens if a user calls this method and then scheduled a new timeout?

A user can schedule a new timeout. I missed that point. Will update it to the `Javadoc`.